assets/chat/js/formatters/UrlFormatter.js

```diff
@@ -71,9 +71,19 @@ export default class UrlFormatter {
       const m = decodedUrl.match(self.linkregex);
       if (m) {
         const encodedUrl = self.encodeUrl(m[0]);
+        const maxUrlLength = 90;
+        let urlText = encodedUrl;
+        if (encodedUrl.length > maxUrlLength) {
+          const middleIndex = Math.floor(maxUrlLength / 2);
+          urlText = `${encodedUrl.slice(
+            0,
+            middleIndex - 3
+          )}...${encodedUrl.slice(middleIndex + 4 - (encodedUrl.length % 2))}`;
```

Comment:
Did you leave this in by mistake?
```suggestion
```

Reply:
Sorry not sure what you're referring to here? feel free to reach out on discord to me to point out specifics Riott#9810

Reply:
It seems like you took my advice in [this comment](https://github.com/destinygg/chat-gui/pull/117/files#r1093697650) and [this comment](https://github.com/destinygg/chat-gui/pull/117/files#r1093712665), but left in the URL-shortening logic from the old PR as well. Looks like you're creating a shortened URL twice.

Reply:
The statement referenced slices the original url to the middle index (accounting for the elipses) adds the elipsis and then slices from after the elipsis to the end accounting for an odd length string. then the line after reduces that further to the first 40 characters each side of the elipsis, keeping the character length the same size either side of the elipsis (at least that was my intent) I very well might just be blind though and have fucked up somewhere 😆

Reply:
looking back a week later it seems pretty hard to read I could push the logic into a function or if we don't really care about perfect placement I can just keep the second line

Reply:
disregard I think I'm stupid, will sort and leave as evidence of my stupidity

Reply:
Yup, I think with my suggested change everything works the same, more or less.

Reply:
Thank you for your patience 🤣